weblab/templates/experiments/experiment_tasks.html

```diff
@@ -32,35 +32,18 @@ <h2>List of active tasks</h2>
                 <p class="failed">You have no queued or running tasks.</p>
             {% endif %}
 
-            {% for experiment in object_list|dictsort:"name" %}
-                {{ experiment.name }}
-                <div class="suppl">
-                    <small>Experiment created
-                        <time>{{ experiment.created_at }}</time>
-                        by <em>{{ experiment.author.full_name }}</em>.
-                    </small>
-                </div>
+            {% for running_experiment in object_list %}
+
                 <ul id="taskList">
-                    {% for version in experiment.versions.all|dictsortreversed:"created_at" %}
-                        <li>
-                            <div id="taskDisplay">
-                                <input class="taskCancelCheckBox" type="checkbox" name="chkBoxes[]"
-                                       value="{{ version.id }}">
-                                <div class="experiment-{{ version.status }}" id="taskDetailBox">
-                                    Version: {{ version.name }} by
-                                    <em>{{ version.author.full_name }}</em>
-                                    <div class="suppl" id="taskVersionDetailBox" style="padding-bottom: 0px">
-                                        <small>Created&nbsp;</small>
-                                        <time>{{ version.created_at }}&nbsp;</time>
-                                        {% with version.files|length as numfiles %}
-                                            <small>containing&nbsp;</small> {{ numfiles }} file{{ numfiles|pluralize }}
-                                        {% endwith %}
-                                    </div>
-                                </div>
+                    <li>
+                        <div id="taskDisplay">
+                            <input class="taskCancelCheckBox" type="checkbox" name="chkBoxes[]"
+                                   value="{{ running_experiment.experiment_version.id }}">
+                            <div class="experiment-{{ running_experiment.experiment_version.status }}" id="taskDetailBox">
+                               {{ running_experiment.experiment_version.experiment.name }} Version: {{ running_experiment.experiment_version.name }}
```

Comment:
As a minor efficiency improvement, you can add the `annotate` I suggested for `expt_name` and use `running_experiment.expt_name` rather than `running_experiment.experiment_version.experiment.name` - it'll save a DB query per experiment_version.

Reply:
Thanks Jonathan. It didn't seem to let me return name using annotation. I got an error message with a list of supportable values none of which was name. I will try again though

Reply:
Ah, of course `Experiment.name` is a Python property not a DB field. Sorry!

Perhaps just `select_related` the experiment as well then?

Reply:
yes thanks, I have added that